src/Cli/dotnet/commands/RestoringCommand.cs

```diff
@@ -99,9 +104,34 @@ private static bool HasArgumentToExcludeFromRestore(IEnumerable<string> argument
 
         private static List<string> FlagsToExcludeFromSeparateRestore =
             ComputeFlags(FlagsToExcludeFromRestore).ToList();
+
+        private static List<string> FlagsThatTriggerSilentSeparateRestore =
+            ComputeFlags(FlagsThatTriggerSilentRestore).ToList();
+
         private static List<string> PropertiesToExcludeFromSeparateRestore =
             ComputePropertySwitches(PropertiesToExcludeFromRestore).ToList();
 
+        private static (string[] newArgumentsToAdd, string[] existingArgumentsToForward) ProcessForwardedArgumentsForSeparateRestore(IEnumerable<string> forwardedArguments)
+        {
+            HashSet<string> newArgumentsToAdd = new();
+            List<string> existingArgumentsToForward = new();
+
+            foreach (var argument in forwardedArguments)
+            {
+
+                if (!IsExcludedFromSeparateRestore(argument) && !IsExcludedFromRestore(argument))
+                {
+                    existingArgumentsToForward.Add(argument);
+                }
+
+                if (TriggersSilentSeparateRestore(argument))
+                {
+                    newArgumentsToAdd.Add("-nologo");
+                    newArgumentsToAdd.Add("-verbosity:quiet");
+                }
+            }
+            return (newArgumentsToAdd.ToArray(), existingArgumentsToForward.ToArray());
+        }
```

Comment:
This got kind of hairy - we needed to check if any getX flag was sent. if so, the separate restore needs to send nologo and quiet. this does lead to an oddity where actual build will send `nologo` too, but that's fine because if `getX` is sent nologo will happen anyway.